client/grpc_query.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	gocontext "context"
+	"fmt"
+
+	gogogrpc "github.com/gogo/protobuf/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+	"google.golang.org/grpc/encoding/proto"
+)
+
+var _ gogogrpc.ClientConn = Context{}
+
+var protoCodec = encoding.GetCodec(proto.Name)
+
+// Invoke implements the grpc ClientConn.Invoke method
+func (ctx Context) Invoke(_ gocontext.Context, method string, args, reply interface{}, _ ...grpc.CallOption) error {
+	reqBz, err := protoCodec.Marshal(args)
+	if err != nil {
+		return err
+	}
+	resBz, _, err := ctx.QueryWithData(method, reqBz)
```

Comment:
nit, should every reply be wrapped with the response height? cc @alexanderbez

Reply:
I'm not sure how that would work here. I believe the response `resBz` already has that info? Correct me if I'm wrong @aaronc